crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -67,11 +67,8 @@ pub trait Environment {
         system: impl AsRef<str> + Send,
     ) -> Result<(), EnvironmentError2>;
 
-    /// Activate this environment
-    async fn activate(&self, nix: &NixCommandLine) -> Result<(), EnvironmentError2>;
-
     /// Atomically edit this environment, ensuring that it still builds
-    async fn edit(&self) -> Result<(), EnvironmentError2>;
+    async fn edit(&self, contents: &impl AsRef<str>) -> Result<(), EnvironmentError2>;
```

Comment:
```suggestion
    async fn edit(&mut self, contents: &impl AsRef<str>) -> Result<(), EnvironmentError2>;
```

can we have this take a mut reference?
I think the transaction requires that and from a type perspective it makes more sense for this to enforce exclusive access semantics